mteb/tasks/BitextMining/multilingual/TedTalksBitextMining.py

```diff
@@ -0,0 +1,178 @@
+    # "eng_Latn": "en",
+    # "epo_Latn": "eo",
+    # "spa_Latn": "es",
+    # "est_Latn": "et",
+    # "eus_Latn": "eu",
+    # "fas_Arab": "fa", # this seems new language
+    # "fin_Latn": "fi",
+    # "fra_Latn": "fr",
+    # "glg_Latn": "gl",
+    # "heb_Hebr": "he",
+    # "hin_Deva": "hi",
+    # "hrv_Latn": "hr",
+    # "hun_Latn": "hu",
+    # "hye_Armn": "hy",
+    # "ind_Latn": "id",
+    # "ita_Latn": "it",
+    # "jpn_Jpan": "ja",
+    # "kat_Geor": "ka",
+    # "kaz_Cyrl": "kk",
+    # "kor_Hang": "ko",
+    # "kur_Arab": "ku", # this seems new language
+    # "lit_Latn": "lt",
+    # "mkd_Cyrl": "mk",
+    # "mon_Cyrl": "mn", # this seems new language
+    # "mar_Deva": "mr",
+    # "msa_Latn": "ms", # this seems new language
+    # "mya_Mymr": "my",
+    # "nob_Latn": "nb",
+    # "nld_Latn": "nl",
+    # "pol_Latn": "pl",
+    # "por_Latn": "pt",
+    # "ron_Latn": "ro",
+    # "rus_Cyrl": "ru",
+    # "slk_Latn": "sk",
+    # "slv_Latn": "sl",
+    # "sqi_Latn": "sq", # this seems new language
+    # "srp_Cyrl": "sr",
+    # "swe_Latn": "sv",
+    # "tam_Taml": "ta",
+    # "tha_Thai": "th",
+    # "tur_Latn": "tr",
+    # "ukr_Cyrl": "uk",
+    # "urd_Arab": "ur",
+    # "vie_Latn": "vi",
+    # "zho_Hans": "zh-cn", # new dialect
+    # "zho_Hant": "zh-tw", # new dialect
+}
+_ISO6391_to_ISO6393 = {v: k for k, v in _ISO6393_to_ISO6391.items()}
+
+_SPLIT = ["train"]
+
+# number of sentences to use for evaluation
+_N = 256
+
+
+def extend_lang_pairs() -> dict[str, list[str]]:
+    # add all possible language pairs
+    hf_lang_subset2isolang = {}
+    for x in _ISO6393_to_ISO6391.keys():
+        if "-" not in x:
+            for y in _ISO6393_to_ISO6391.keys():
+                if x != y:
+                    pair = f"{x}-{y}"
+                    hf_lang_subset2isolang[pair] = [
+                        x.replace("_", "-"),
+                        y.replace("_", "-"),
+                    ]
+    return hf_lang_subset2isolang
+
+
+_ISO6393_to_ISO6391_MAPPING = extend_lang_pairs()
+
+
+class TedTalksBitextMining(AbsTaskBitextMining, CrosslingualTask):
+    metadata = TaskMetadata(
+        name="TedTalksBitextMining",
+        dataset={
+            "path": "davidstap/ted_talks",
+            "revision": "e840d90795ba7f40dfad0a51ecb6e244840ebdd5",
+            "trust_remote_code": True,
+        },
+        description="Collection of multitarget bitexts based on TED Talks",
+        reference="https://huggingface.co/datasets/davidstap/ted_talks",
+        type="BitextMining",
+        category="s2s",
+        eval_splits=_SPLIT,
+        eval_langs=_ISO6393_to_ISO6391_MAPPING,
+        main_score="f1",
+        date=None,
+        form=None,
+        domains=["Spoken"],
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        n_samples={"train": _N},
+        avg_character_length={},
```

Comment:
these need to be filled out - documentation can be found in taskmetadata object or in the "adding a dataset"-guide.